### PR TITLE
Fix issue #2778: move QDir::toNativeSeparators() to helper functions

### PR DESCRIPTION
**Pull request by Aaron19960821**
This pr aims to drive #2778 .
We will convert filepaths() to native separators in base functions we provide.

### CODE REVIEW

File: src/appleseed.qtcommon/utility/miscellaneous.cpp [hunk 1]
```diff
@@ -276,10 +276,11 @@ QString get_open_filename(
         settings,
         settings_key + SettingsLastDirectory,
         QDir::toNativeSeparators(QFileInfo(filepath).path()));
-
-    return filepath;
+        
+    return QDir::toNativeSeparators(filepath);
 }
 
+
```
Comment (oktomus) [inline src/appleseed.qtcommon/utility/miscellaneous.cpp]: Remove empty line

File: src/appleseed.studio/mainwindow/project/texturecollectionitem.cpp [hunk 2]
```diff
@@ -141,13 +141,10 @@ void TextureCollectionItem::slot_import_textures()
...
         return;
 
-    const bf::path path(
-        QDir::toNativeSeparators(filepaths.first()).toStdString());
-
     // todo: schedule creation of texture and texture instances when rendering.
     for (int i = 0; i < filepaths.size(); ++i)
     {
-        const std::string filepath = QDir::toNativeSeparators(filepaths[i]).toStdString();
+        const std::string filepath(filepaths[i].toStdString());
```
Comment (oktomus) [inline src/appleseed.studio/mainwindow/project/texturecollectionitem.cpp]: Elsewhere in appleseed, we prefer to write it this way (easier to read) :
```cpp
const std::string filepath = filepaths[i].toStdString();
```

File: src/appleseed.qtcommon/utility/miscellaneous.cpp [hunk 3]
```diff
@@ -310,6 +311,11 @@ QStringList get_open_filenames(
         settings_key + SettingsLastDirectory,
         QDir::toNativeSeparators(QFileInfo(filepaths.first()).path()));
 
+    for (int i = 0; i < filepaths.size(); ++i) 
+    {
+        filepaths[i].replace(0, filepaths[i].length(), QDir::toNativeSeparators(filepaths[i]));
```
Comment (oktomus) [inline src/appleseed.qtcommon/utility/miscellaneous.cpp]: That looks fishy, can't you do this ?
```cpp
filepaths[i] = QDir::toNativeSeparators(filepaths[i]);
```

Comment (Aaron19960821) [reply src/appleseed.qtcommon/utility/miscellaneous.cpp]: Thanks for your review, all issues have addressed.

**Pull request merged**